### ISSUE TITLE
Borg erroneously thinks my files have changed

### ISSUE THREAD

**Issue by auanasgheps** [borgbackup/borg#5770]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

Issue

#### Your borg version (borg -V).
borg 1.1.15

#### Operating system (distribution) and version.
Debian 10 / OMV5

#### Hardware / network configuration, and filesystems used.
Source: Debian 10 server, ext4, borg 1.1.15
Destination: remote Synology NAS running borg 1.1.13

#### How much data is handled by borg?

4 TB (uncompressed, deduplicated)

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

Please note I am using borgmatic to automate. The actual executed command is:

`borg create --exclude-from /tmp/tmp410uy2bb --compression zstd --remote-ratelimit 7300 --remote-path /usr/local/bin/borg --list --filter AME- --info MYUSER@MYSERVER:backup::{hostname}-{now:%Y-%m-%dT%H:%M:%S} /srv/dev-disk-by-label-nvme /srv/fb4ab0da-2b79-4740-834a-7435c0123e28`

## Describe the problem you're observing.
Hi, I've been happily running backups with borg every week for last 4 months from my local server to a remote Synology NAS.

Last run (on Saturday) did not end when I expected, and I had to stop it. Most of my data does not change every week, and even if I do big stuff in 3 to 4 hours it's done.

When enabling bormatic file ouput, I discovered that borg thinks my files have been modified: there's an M at the beginning of every entry.

They have not.

In fact I also run a rsync job every week on a USB hard drive, and the transfer was quite short, since almost nothing changed from the week before. I believe rsync uses date/time to check if a file has changed. Either way, I'm 100% those files haven't been touched.

This is the first time I'm seeing the issue since I've setup borg. 

Since I had this issue the backup has been running for about 20 hours, but hasn't completed yet. It's not a problem of bandwidth: most of the time is spent reading non-changed local files that borg thinks have changed.

After the upload of the actual changed files is dove, borg is only uploading these files:

![immagine](https://user-images.githubusercontent.com/20586878/115275789-8965dd80-a142-11eb-9929-acb26e57323b.png)

I can't afford to wait for borg to read **ALL**  of my data again!

I haven't done any other change to borg client, borg server or my data.

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

Yes, since the last full backup. I've started the backup multiple times with the same result.

**Comment by auanasgheps** [comment]
> Did you read [FAQ It always chunks all my files, even unchanged ones!](https://borgbackup.readthedocs.io/en/stable/faq.html#it-always-chunks-all-my-files-even-unchanged-ones)?

Yes I did, but could not understand much more than "the cache will end after a number of backups". I don't think I have reached 20 runs, but still, it doesn't make much sense to me.



> * borg uses (by default) ctime/inode/size - this is 100% safe (AFAIK), but slower for some metadata-only changes (like chmod/chown), because they will update the ctime although the file content did not change. similar if you copy all files 1:1 to a new filesystem, in that case the inode will change.

Can you please tell/reference how to verify inode? I will check for crime but I'm 99.99% hasn't changed along with the size.

**Comment by ThomasWaldmann** [comment]
If you can upgrade to 1.1.16 on the client, you can use this (citing from changelog):
```
verbose files cache logging via --debug-topic=files_cache, #5659.
Use this if you suspect that borg does not detect unmodified files as expected.
```

**Comment by auanasgheps** [comment]
> If you can upgrade to 1.1.16 on the client, you can use this (citing from changelog):
> 
> ```
> verbose files cache logging via --debug-topic=files_cache, #5659.
> Use this if you suspect that borg does not detect unmodified files as expected.
> ```

Okay, I'll have to do this manually since the update is not yet in the Debian repo.

**Comment by auanasgheps** [comment]
Hey, do I have other methods to troubleshoot? I can't upgrade borgbackup to 1.1.16 due to non updated dependencies (phython3 and libzstd1). I don't know if they're used by other software and don't want to break stuff.

**Comment by infectormp** [comment]
In issue description you say "ext4" now you say mergeFS. This is a big difference.

https://borgbackup.readthedocs.io/en/stable/faq.html#i-am-having-troubles-with-some-network-fuse-special-filesystem-why
> if you are encountering issues like slowness, corruption or malfunction when using a specific filesystem, please try if you can reproduce the issues with a local (non-network) and proven filesystem (like ext4 on Linux).

**Comment by auanasgheps** [comment]
Sorry, ext4 is the actual filesystem of my disks, I didn't think mergeFS could make a difference. I could point to the drives directly, but this would invalidate the cache anyway since the path would change.

**Comment by ThomasWaldmann** [comment]
You need to tell the filesystem from the perspective of borg (software), not from the perspective of your disk hw.

So, if you let borg backup files on a mergeFS mount, the fs is mergeFS.

**Comment by auanasgheps** [comment]
I apologize for the misunderstanding. 
By the way, the borg backup has always run on MergeFS and I've never experienced a real issue or slowdown. 

I could open an issue to MergeFS, but files and their properties haven't changed at all so I don't know what I should complain to them. 

What else can be done without upgrading to 1.1.16, which requires dependencies to be updated on my system?

**Comment by auanasgheps** [comment]
I've reached out to the MergerFS dev, who very kindly explained what probably has happened.

![image](https://user-images.githubusercontent.com/20586878/115591325-43d81a80-a2d2-11eb-8179-14d7f739de3c.png)

Therefore I will disable _inode_ from cache calculation to avoid future disruption. I hope is still "safe" keeping ctime/size.

Maybe you could consider adding this workaround to [this FAQ](https://borgbackup.readthedocs.io/en/stable/faq.html#i-am-having-troubles-with-some-network-fuse-special-filesystem-why)? 
I hope nobody else in the future faces this issue :)

Thanks for the help, I've learned something today as well!

**Comment by jdchristensen** [comment]
I think this FAQ entry is more relevant:
https://borgbackup.readthedocs.io/en/stable/faq.html#it-always-chunks-all-my-files-even-unchanged-ones
The one about different file systems could link to it.

One complaint about the one about rechunking is that the ...TTL variable is mentioned early on, but it is rarely the issue.  I think the issues with unstable inodes, paths, etc. should be mentioned earlier and more prominently.

(BTW, you can run the standalone borg binary if you want to test this further, as that has needed dependencies packaged with it, so it doesn't need to be installed.)

**Comment by auanasgheps** [comment]
Hi,
Just for reference: I disabled inode cache and was able to continue my backup as usual. 

This can be closed for me, but I would like to see this kind of recommendation in one of the two FAQs mentioned above.
